pkg/expr/classic/classic.go

```diff
@@ -11,52 +11,44 @@ import (
 	"github.com/grafana/grafana/pkg/expr/mathexp"
 )
 
-// ConditionsCmd is command for the classic conditions
-// expression operation.
-type ConditionsCmd struct {
+// ClassicConditionsCmd is a command that supports the reduction and comparison of conditions.
+//
+// A condition in ClassicConditionsCmd can reduce a time series; or contain either an instant metric
+// or the result of another expression; and check if it exceeds a threshold, falls within a
+// range, or does not contain a value.
+//
+// If ClassicConditionsCmd contains more than one condition, it reduces the boolean outcomes of the
+// threshold, range or value checks using the logical operator of the right hand side condition
+// until all conditions have been reduced to a single boolean outcome. ClassicConditionsCmd does not
+// follow operator precedence.
+//
+// For example if we have the following classic condition:
+//
+//	min(A) > 5 OR max(B) < 10 AND C = 1
```

Comment:
Might make sense to call out which parts of this structure are reducers, operators, and evaluators, and which conditions each of them belong to:

```
   | reducer           | operator           | query reference
// min(A) > 5         (OR max(B) < 10) (AND C = 1)
   | condition 0      | condition 1    | condition 2

```